File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int n;

    // prompt for positive height
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);

    for (int i = 0; i < n; i++)
    {
        // print spaces first
        for (int j = 0; j < n - i - 1; j++)
        {
            printf(" ");
        }
        // print hashes
        for (int k = 0; k < i + 1; k++)
        {
            printf("#");
        }
        // print spaces
        printf("  ");
        // print other side hashes (spaces not needed)
        // print hashes
        for (int l = 0; l < i + 1; l++)
        {
            printf("#");
        }
        // newline
        printf("\n");
    }
}

File: pset6/mario/more/mario.py
from cs50 import get_int


def main():
    height = get_height()
    for i in range(height):
        print(" " * (height - i - 1), end="")
        print("#" * (i + 1), end="")
        print(" " * 2, end="")
        print("#" * (i + 1))


def get_height():
    while True:
        h = get_int("Height: ")
        if h > 0 and h < 9:
            break
    return h


main()